### DISCUSSION TITLE
UseRouter hook not functioning

### DISCUSSION THREAD

**Discussion by Revaycolizer** [vercel/next.js#48374] [Help]
### Summary

Hi guys I'm unable to perform routing I don't know why but below is how I implemented it in appDir nextjs 13.2. 4
```
"use client" 
import {useRouter} from 'next/navigation'

const router = useRouter()
const Login=() =>{
router.push("/home")
}
```

**Comment by Dio-brandio** [comment]
useRouter doesn't work in  server side componetns

**Reply by Revaycolizer** [reply]
I am using it inside a client component as I included "use client"

**Reply by 808vita** [reply]
That person is right "useRouter" will not work, you need to follow the hook rules of react .

Instead make use of Router from "next/router"  this worked for me .
Attaching links to utils I have used.

[Router push ](https://github.com/808vita/nextjs-blog-with-auth/blob/deploy/src/util/routerPush.tsx)

[Router reload](https://github.com/808vita/nextjs-blog-with-auth/blob/deploy/src/util/routerReload.tsx)

[Router replace](https://github.com/808vita/nextjs-blog-with-auth/blob/deploy/src/util/routerReplace.tsx)

```
import Router from "next/router";

export const routerPush = (path: string) => {
  if (path === "") {
    Router.push("/");
  }

  Router.push(path);
};

```
Just import the function and call it with your required path .
example:
` routerPush("/home") ;`
this will route to "/home"

**Reply by 808vita** [reply]
ps. I have not tried it in appdir

**Reply by Revaycolizer** [reply]
I already solved it bruh I I added
```
const handlesubmit=(e:React.FormEvent)=>{
...
}
```
It worked

**Reply by stjepanoviccc** [reply]
@Revaycolizer how did you solve it? I'm using cart on session, and since i have some filters on my shop, i need to store one of category filter inside of localStorage**(when i call from outside of shop page)** because my route is /shop not /shop/category...
And when  do window.location.href it will refresh page and lose cart context, but when i try to push with router it keep getting errors: _You should only use "next/router" on the client side of your app._

```
"use client";
import Router from "next/router";

export const routerPush = (path) => {
  if (typeof window !== "undefined") {
    if (path === "") {
      Router.push("/");
    } else {
      Router.push(path);
    }
  }
};
```
```

**Reply by Revaycolizer** [reply]
You should import useRouter from next/navigation 

And do something like

```
const router = useRouter()

router.push(`/shop?page=$yourValue`)

**Reply by Revaycolizer** [reply]
You should import useRouter from next/navigation 

And do something like

```
const router = useRouter()

router.push(`/shop?page=${yourValue}`)

```
